### PR TITLE
Job Improvement Project - Sura

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: #4210

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * kRO Changelog: http://ro.gnjoy.com/news/notice/View.asp?seq=7099
Thanks to @flamefury and Sigma!

### CODE REVIEW

**Comment by attackjom** [comment]
rampage blast still consume all of spirit sphere of user 
I think it need to comment or remove case Rampageblaster  in skill.cpp

File: src/map/battle.cpp [hunk 1]
```diff
@@ -4033,14 +4042,14 @@ static int battle_calc_attack_skill_ratio(struct Damage* wd, struct block_list *
 			break;
 		case SR_EARTHSHAKER:
 			if (tsc && ((tsc->option&(OPTION_HIDE|OPTION_CLOAK|OPTION_CHASEWALK)) || tsc->data[SC_CAMOUFLAGE] || tsc->data[SC_STEALTHFIELD] || tsc->data[SC__SHADOWFORM])) {
-				//[(Skill Level x 150) x (Caster Base Level / 100) + (Caster INT x 3)] %
-				skillratio += -100 + 150 * skill_lv;
+				//[(Skill Level x 300) x (Caster Base Level / 100) + (Caster STR x 3)] %
```
Comment (attackjom) [inline src/map/battle.cpp]: from this http://ro.gnjoy.com/news/update/View.asp?seq=235&curpage=1  and this https://www.divine-pride.net/database/skill/2328/

It seem the new base damage formula is
-> Normal target  -> -100 + 300 * skill_lv

-> Hidden target -> -100 + 400* skill_lv

File: src/map/battle.cpp [hunk 2]
```diff
@@ -4066,36 +4075,41 @@ static int battle_calc_attack_skill_ratio(struct Damage* wd, struct block_list *
...
+			//ATK [{(Skill Level x 80) + (Caster AGI)} x Caster Base Level / 100] %
+			skillratio += -100 + 80 * skill_lv + sstatus->agi;
 			RE_LVL_DMOD(100);
 			break;
 
 		case SR_RAMPAGEBLASTER:
-			if (sc && sc->data[SC_EXPLOSIONSPIRITS]) {
-				skillratio += -100 + (20 * sc->data[SC_EXPLOSIONSPIRITS]->val1 + 20 * skill_lv) * ((sd) ? sd->spiritball_old : 1);
+			if (tsc && tsc->data[SC_EARTHSHAKER]) {
+				skillratio += -100 + 550 * skill_lv;
```
Comment (attackjom) [inline src/map/battle.cpp]: New formula is

Normal target -> 900 + 350 * skill_lv

Earthshaker state target -> 1400 + 550 * skill_lv

**Comment by attackjom** [comment]
Tiger cannon skill   need to change formula 

from using abs(skill_get_hp//sp_rate(skill_id, skill_lv))  to 

hp = sstatus->max_hp * (12+(skill_lv*2))  / 100,
sp = sstatus->max_sp * (5+skill_lv) / 100;

File: src/map/battle.cpp [hunk 3]
```diff
@@ -4118,7 +4134,7 @@ static int battle_calc_attack_skill_ratio(struct Damage* wd, struct block_list *
 			break;
 		case SR_RIDEINLIGHTNING: // ATK [{(Skill Level x 200) + Additional Damage} x Caster Base Level / 100] %
 			skillratio += -100 + 200 * skill_lv;
-			if ((sstatus->rhw.ele) == ELE_WIND || (sstatus->lhw.ele) == ELE_WIND)
+			if (sd && sd->status.weapon == W_KNUCKLE)
```
[the remark this answers is not in the document]
Comment (aleos89) [reply src/map/battle.cpp]: Where do you see it was changed to 25%?

[the remark this answers is not in the document]
Comment (attackjom) [reply src/map/battle.cpp]: In the link in my comment then first skill

[the remark this answers is not in the document]
Comment (attackjom) [reply src/map/battle.cpp]: The formula of bonus damage is 1.25x relative to normal damage

**Comment by ecdarreola** [comment]
SR_KNUCKLEARROW should be level 10.

Also, if you change it to level 10 in skill_tree and test it in-game. There are cells that the caster is not moving to the target. (e.g. below)

image

File: src/map/battle.cpp [hunk 4]
```diff
@@ -4064,8 +4064,8 @@ static int battle_calc_attack_skill_ratio(struct Damage* wd, struct block_list *
  			break;
 		case SR_TIGERCANNON:
 			{
-				unsigned int hp = sstatus->max_hp * abs(skill_get_hp_rate(skill_id, skill_lv)) / 100,
-							 sp = sstatus->max_sp * abs(skill_get_sp_rate(skill_id, skill_lv)) / 100;
+				unsigned int hp = sstatus->max_hp * (12 + (skill_lv * 2)) / 100,
+							 sp = sstatus->max_sp * (5 + skill_lv) / 100;
```
Comment (llchrisll) [inline src/map/battle.cpp]: Shouldn't it be 6% per Skill Level?
`[Level 1] : 범위 3 x 3셀 / MHP의 12% / MSP의 6%
[Level 2] : 범위 3 x 3셀 / MHP의 14% / MSP의 7%
[Level 3] : 범위 3 x 3셀 / MHP의 16% / MSP의 8%
[Level 4] : 범위 3 x 3셀 / MHP의 18% / MSP의 9%
[Level 5] : 범위 3 x 3셀 / MHP의 20% / MSP의 10%
[Level 6] : 범위 5 x 5셀 / MHP의 22% / MSP의 11%
[Level 7] : 범위 5 x 5셀 / MHP의 24% / MSP의 12%
[Level 8] : 범위 5 x 5셀 / MHP의 26% / MSP의 13%
[Level 9] : 범위 5 x 5셀 / MHP의 28% / MSP의 14%
[Level 10] : 범위 5 x 5셀 / MHP의 30% / MSP의 15%`

**Comment by AceWindstorm** [comment]
Double checking that Tiger Cannon still is supposed to consume HP? 

image

Taken from : http://ro.gnjoy.com/news/devnote/View.asp?category=1&seq=3935883&curpage=1

Translation here is far less ambiguous than on the changelog link from the first post. It is also more current, being posted on 5/20, while the other was on 4/15.

**Comment by someonenoone11** [comment]
Sura skill tree is all wrong. And it's responsible for the majority of my confusion atm.

List view of skills, everything works fine.

Skill tree view however, it will highlight the wrong things as requirements, and spend skill points into those wrong things. For example, if i want to increase Rising Dragon in tree view, it will put points into PP energy and rampage blaster.

The work around is simply using the list view. But it's still an issue.

**Comment by llchrisll** [comment]
> Sura skill tree is all wrong. And it's responsible for the majority of my confusion atm.
> 
> List view of skills, everything works fine.
> 
> Skill tree view however, it will highlight the wrong things as requirements, and spend skill points into those wrong things. For example, if i want to increase Rising Dragon in tree view, it will put points into PP energy and rampage blaster.
> 
> The work around is simply using the list view. But it's still an issue.

Did you update your client lua files?

**Comment by someonenoone11** [comment]
> Did you update your client lua files?
> It should look like this:
> If you get my translation project, continued from zackdreaver's where I did the Sura Update already:
> https://github.com/llchrisll/ROenglishRE
> You would be fine ^^.

It looks like my noobiness is doing more harm than good. 2/3 issues i've submitted were noob mistakes not issues : (

Thank you very much! Everything is fine. I will delete this comment in 24hrs to not clutter up this page.

Thank you.

**Comment by Dieby** [comment]
3.1. Arm Cannon
- Reduces fixed casting time from 0.2 seconds to 0.1 seconds based on level 5.
- Reduces variable casting time from 2.2 seconds to 2 seconds based on level 5.
- Reduces cooldown from 0.65 seconds to 0.3 seconds based on level 5.
- - Improves damage formula. New damage formula : (400 + (300 x skill level))%Atk regardless of target size, improves base level modifier from (base level/120) to (base level/100).
### - Cast range is 9 cells regardless of skill level. Area of effect : 3x3 cells (level 1 ~ 3) / 5x5 cells (level 4 ~ 5).

range: 9 only take rank at lvl 5

could anyone solve this?

**Pull request merged**